cpp/src/unary/math_ops.cu

```diff
@@ -209,9 +209,36 @@ struct DeviceRInt {
   }
 };
 
-// bitwise op
+struct DeviceBitCount {
+  template <typename T>
+  int32_t __device__ operator()(T data)
+  {
+    auto constexpr nbits = CHAR_BIT * sizeof(T);
+    if constexpr (nbits <= 32) {
+      return __popc(static_cast<unsigned int>(data));
+    } else if constexpr (nbits <= 64) {
+      return __popcll(static_cast<unsigned long long>(data));
+    } else if constexpr (nbits == 128 && sizeof(unsigned long long) == 64) {
+      // Size of unsigned long long is not guaranteed to be 64 bits,
```

Comment:
If sizeof(unsigned long long) is not 64, may be we can split this computation in 4 x ints with `__popc`?

Side note: In C++20, there is a `std::popcount` (https://en.cppreference.com/w/cpp/numeric/popcount), not sure if we will have a `cuda::std::popcount` but if yes, we can add a todo here to use that instead as well.

Reply:
> may be we can split this computation in 4 x ints with __popc?

Calling it 4X seems not worth the hassle. I'm adding a comment on the `std::popcount`.

Reply:
Wait, `std::popcount` is C++20. Checking if that is already available in device code....

Reply:
Done.